controllers/runnermanager.go

```diff
@@ -281,65 +273,67 @@ func difference(prev, current []string) []string {
 	return ret
 }
 
-func (m *managerLoop) notifyToSlack(ctx context.Context, runnerList []*github.Runner, podList *corev1.PodList) error {
-	for i := range podList.Items {
-		po := &podList.Items[i]
-		jobResult, err := m.runnerPodClient.GetJobResult(ctx, po.Status.PodIP)
-		if err != nil {
-			m.log.Error(err, "skipped notification because failed to get the job result from the runner pod API", "pod", namespacedName(po.Namespace, po.Name))
-			continue
-		}
-
-		if jobResult.Status == rc.JobResultUnfinished {
-			m.log.Info("skipped notification because pod is not finished", "pod", namespacedName(po.Namespace, po.Name))
-			continue
-		}
-		if jobResult.FinishedAt.Before(m.lastCheckTime) {
-			m.log.Info("skipped notification because pod status is not updated", "pod", namespacedName(po.Namespace, po.Name))
-			continue
-		}
-
-		if len(m.slackAgentServiceName) != 0 {
-			m.log.Info("send an notification to slack-agent", "pod", namespacedName(po.Namespace, po.Name))
-			c, err := agent.NewClient(fmt.Sprintf("http://%s", m.slackAgentServiceName))
-			if err != nil {
-				return err
-			}
-			return c.PostResult(ctx, m.slackChannel, jobResult.Status, *jobResult.Extend, po.Namespace, po.Name, jobResult.JobInfo)
-		} else {
-			m.log.Info("skip sending a notification to slack-agent because service name is blank", "pod", namespacedName(po.Namespace, po.Name))
-		}
-		return nil
+func (m *managerLoop) notifyToSlack(ctx context.Context, po *corev1.Pod, status *runner.Status) error {
+	// FIXME: create this client in StartOrUpdate()
+	c, err := agent.NewClient(fmt.Sprintf("http://%s", m.slackAgentServiceName))
+	if err != nil {
+		return err
 	}
-	return nil
+	return c.PostResult(ctx, m.slackChannel, status.Result, *status.Extend, po.Namespace, po.Name, status.JobInfo)
 }
 
 func (m *managerLoop) maintainRunnerPods(ctx context.Context, runnerList []*github.Runner, podList *corev1.PodList) error {
 	now := time.Now().UTC()
+	lastCheckTime := m.lastCheckTime
+	m.lastCheckTime = now
```

Comment:
I think `m.lastCheckTime` should be updated if `m.notifyToSlack` in line 320 succeeds.

Reply:
That's good. However, that may result in duplicate slack notifications.
So I'd like to save the result of the slack notification as an annotation.

https://github.com/cybozu-go/meows/issues/96